pkg/osv/osv_test.go

```diff
@@ -0,0 +1,50 @@
+package osv
+
+import (
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+)
+
+func TestRetryOn5xx(t *testing.T) {
```

Comment:
issue: we should be testing for a couple of different status codes and orders

I'd recommend refactoring this into a table-based test, and we should have cases for at least:
  - 200
  - 4xx
  - 5xx
  - 5xx, then a 2xx